types/integer/index.d.ts

```diff
@@ -0,0 +1,107 @@
+// Type definitions for integer 1.0
+// Project: https://github.com/JoshuaWise/integer#readme
+// Definitions by: Ben Davies <https://github.com/Morfent>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+// This can't just be written as a simple class since Integer instances can't
+// be constructed by its actual constructor without throwing, but calling it as
+// a regular function constructs new instances just fine...
+declare function Integer(val: Integer.IntLike): Integer.IntClass;
+
+declare namespace Integer {
+	type IntLike = number | string | IntClass;
+
+	function fromBits(low: number, high?: number): IntClass;
+	function fromNumber(val: number, defaultVal?: IntClass): IntClass;
+	function fromString(val: string, radix?: number): IntClass;
```

Comment:
There's an optional third argument (to specify a default value) which can either be a `string` or `IntClass`